### PR TITLE
Fixed test `test_static_cache_exportability` with torch 2.4.0

### PR DESCRIPTION
**Pull request by guangy10**
# What does this PR do?

A workaround to run test `test_static_cache_exportability` with torch 2.4.0.

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. Issue discussed in the channel.
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@gante

### CODE REVIEW

**Review by gante — APPROVED** [review]
LGTM, thank you for the fix 💛

**Review by amyeroberts — COMMENTED** [review]
Thanks for fixing! 

Just a question on the compatible version

File: tests/utils/test_cache_utils.py [hunk 1]
```diff
@@ -174,6 +174,8 @@ def test_static_cache_exportability(self):
         """
         Tests that static cache works with `torch.export()`
         """
+        import torch.export._trace
```
[the remark this answers is not in the document]
Comment (gante) [reply tests/utils/test_cache_utils.py:177]: @amyeroberts should we gate against older versions on tests as well? (If so, I'm sure we have a couple of tests in need of gating, e.g. [this one](https://github.com/huggingface/transformers/blob/b51d4145bbfa9bd90dc878e552d5cdc5133dabc2/tests/generation/test_utils.py#L1855) :D)

[the remark this answers is not in the document]
Comment (amyeroberts) [reply tests/utils/test_cache_utils.py:177]: Technically yes as anyone can should be able to run the test suite. 

It's not a huge deal if it's not already enabled + nobody has complained - our runners use torch 2.4 so it won't break things if it available from that version onwards. 

In the case of this test, we already have version gating (I'm assuming for a certain reason that was flagged) so probably best to remain conservative.

**Comment by guangy10** [comment]
Fixed import compatible version. Should be good to go.

**Review by amyeroberts — APPROVED** [review]
Thanks!

**Pull request merged**